Source/main.cpp

```diff
@@ -499,46 +609,81 @@ void main_main ()
-	    
 	    outputFile << "time = " << time << " "
                     << "Sum_normalized_M: "
                     << normalized_Mx/num_mag << " "
                     << normalized_My/num_mag << " "
                     << normalized_Mz/num_mag << std::endl;
 
-             outputFile << "time = " << time << " "
+            // Check if field is equilibirated
+	    // If so, we will increment 
+            if ((Hbias_sweep == 1) && (step > 1)) {
+	    
+	        err_x = amrex::Math::abs((normalized_Mx/num_mag) - normalized_Mx_old);
+	        err_y = amrex::Math::abs((normalized_My/num_mag) - normalized_My_old);
+	        err_z = amrex::Math::abs((normalized_Mz/num_mag) - normalized_Mz_old);
+                 
+		outputFile << "time = " << time << " "
+                    << "error: "
+                    << err_x << " "
+                    << err_y << " "
+                    << err_z << std::endl;
+
+
+ 	        normalized_Mx_old = normalized_Mx/num_mag;
+	        normalized_My_old = normalized_My/num_mag;
+	        normalized_Mz_old = normalized_Mz/num_mag;
+
+	        if ((err_x < equilibrium_tolerance) && (err_y < equilibrium_tolerance) && (err_z < equilibrium_tolerance)) {
+	            increment_Hbias = 1;
+
+                    // Reset the error
+	            normalized_Mx_old = 0.;
+		    normalized_My_old = 0.;
+                    normalized_Mz_old = 0.;
+		}
+
+	    }
+
+            // standard problem 2 diagnostics
+	    if (diag_type == 2) {
+
+                Real Heff_x = SumHeff(H_demagfield[0], H_exchangefield[0], H_biasfield[0]);
+                Real Heff_y = SumHeff(H_demagfield[1], H_exchangefield[1], H_biasfield[1]);
+                Real Heff_z = SumHeff(H_demagfield[2], H_exchangefield[2], H_biasfield[2]);
+
+	        Real dot_product = normalized_Mx/num_mag + normalized_My/num_mag + normalized_Mz/num_mag;
+	    
+	        outputFile << "time = " << time << " "
                     << "Heff: "
                     << Heff_x/num_mag << " "
                     << Heff_y/num_mag << " "
                     << Heff_z/num_mag << std::endl;
             
-	     outputFile << "time = " << time << " " 
+	        outputFile << "time = " << time << " " 
 		    << "dot_product = " << dot_product << std::endl;
  
 
-             if (dot_product < 0 && dot_product_prev >= 0.){
-	         Real Heff_magn = sqrt(Heff_x*Heff_x + Heff_y*Heff_y + Heff_z*Heff_z);
-
-		 outputFile << "time = " << time << " "
-                    << "dot_product_zero_magnitude = " << Heff_magn  <<  std::endl;
-	     }	     
+                if ((dot_product < 0 && dot_product_prev >= 0.) || (dot_product > 0 && dot_product_prev <=0.)){
+	           Real Heff_magn = sqrt(Heff_x*Heff_x + Heff_y*Heff_y + Heff_z*Heff_z);
 
-	     dot_product_prev = dot_product;
+		   outputFile << "time = " << time << " "
+                              << "dot_product_zero_magnitude = " << Heff_magn  <<  std::endl;
```

Comment:
@jclepelch I think this should be the magnitude of H_bias, not the total H_eff (not 100% sure...)